### PR TITLE
Validate chainId when accessing faucet or contract deployment

### LINKED ISSUE #95: Faucet & deploy should check the chain id on the web3 provider
And prompt the user to switch to the ewasm testnet (+ show the RPC url or a link to ewasm/testnet).

### PR DESCRIPTION
**Pull request by hugo-dc**
closes https://github.com/ewasm/etherchain-light/issues/95

It is OK to allow deploying a contract, or request faucet funds if metamask is not connected to the ewasm testnet?

### CODE REVIEW

File: public/js/utils.js [hunk 1]
```diff
@@ -89,3 +89,10 @@ function submitTransaction () {
     }
   })
 }
+
+function validateChainId() {
+  var chainId = metamask.version.network;
+  if (chainId !== "66") {
```
Comment (axic) [inline public/js/utils.js]: I think this should go into the configuration.

**Comment by axic** [comment]
> It is OK to allow deploying a contract, or request faucet funds if metamask is not connected to the ewasm testnet?

How can it deploy a contract in that case?

Requesting money should work, but to avoid confusion I'd say both cases must require the correct chain id.

**Comment by hugo-dc** [comment]
> How can it deploy a contract in that case?

It won't deploy the contract, but it would try to (it would open metamask and probably metamask will show an error)

I'll add another message when the user press the deploy button if network is incorrect, and avoiding calling metamask.

File: public/js/utils.js (hunk 1 above)
Comment (hugo-dc) [reply public/js/utils.js]: chainId now can be configured in `config.js`

File: public/js/utils.js [hunk 2]
```diff
@@ -89,3 +89,10 @@ function submitTransaction () {
     }
   })
 }
+
+function validateChainId(chainId) {
+  var currId = metamask.version.network;
+  if (currId !== chainId) {
+    alert("WARNING: Metamask is connected to a different Ethereum network\nPlease configure correct RPC url for the ewasm testnet:\nhttp://ewasm.ethereum.org:8545");
```
Comment (axic) [inline public/js/utils.js]: I think the configuration could have a field for either:
a) `rpcUrlSuggestion` - if it is non-empty, then the "Please configure correct RPC..." text is displayed with it
b) `invalidChainConnected` to contain the entire message displayed here

File: views/faucet.pug [hunk 3]
```diff
@@ -78,19 +78,9 @@ block content
...
-          await window.ethereum.enable();
-        } catch (error) {
-          console.log(error);
-        }
-      }
-    }
-
     window.addEventListener('load', async () => {
       await enableEthereum();
+      validateChainId(!{JSON.stringify(data.chainId)});
```
Comment (axic) [inline views/faucet.pug]: What is the `!{` part doing?

File: public/js/utils.js (hunk 2 above)
Comment (hugo-dc) [reply public/js/utils.js]: added `rpcUrlSuggestion` to configuration

File: views/faucet.pug (hunk 3 above)
Comment (hugo-dc) [reply views/faucet.pug]: `!{` is for unescaped strings, in this case the rendered js will look like:
```js
validateChainId("66");
```
using `#{` would generate this invalid js code:
```js
validateChainId(&quot;66&quot;)
```
```

File: views/faucet.pug [hunk 4]
```diff
@@ -77,7 +77,6 @@ block content
         window.location.hash = "#transaction";
       }
     }
-
     async function enableEthereum() {
```
Comment (axic) [inline views/faucet.pug:80]: @hugo-dc added this back here, because not sure it should be removed. But if it does, I think it should be done in a separate PR.
